src/routes/safe/components/Transactions/TxsTable/index.tsx

```diff
@@ -83,58 +83,65 @@ const TxsTable = ({ classes }) => {
           size={filteredData.size}
         >
           {(sortedData) =>
-            sortedData.map((row, index) => (
-              <React.Fragment key={index}>
-                <TableRow
-                  className={cn(classes.row, expandedTx === row.tx.safeTxHash && classes.expandedRow)}
-                  data-testid={TRANSACTION_ROW_TEST_ID}
-                  onClick={() => handleTxExpand(row.tx.safeTxHash)}
-                  tabIndex={-1}
-                >
-                  {autoColumns.map((column: any) => (
-                    <TableCell
-                      align={column.align}
-                      className={cn(classes.cell, ['cancelled', 'failed'].includes(row.status) && classes.cancelledRow)}
-                      component="td"
-                      key={column.id}
-                      style={cellWidth(column.width)}
-                    >
-                      {row[column.id]}
+            sortedData.map((row, index) => {
+              const rowCombinedId = `${row.tx.nonce + row.tx.data}`
```

Comment:
why did we have to switch from `safeTxHash` to this? This doesn't look unique
Is that because of creation tx? Anyways we need to find a better solution

Reply:
I don't like this either, and agree with the "better solution".

The problem is with the _mocked_/_pending_ txs. I couldn't persist their state of expanded/collapsed between updates, as the `safeTxHash` wasn't available at the time of the tx creation.

Reply:
When you approve a transaction you approve a hash (this one is `safeTxHash`) and you know it before calling `.approveHash`, not sure how we can get one in case of execution

Reply:
But I'm 100% sure that Richard knows an answer to this question